--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -6017,6 +6017,20 @@
 			_("YES"), [] { Utils::Platform::quitES(Utils::Platform::QuitMode::REBOOT); },
 			_("NO"), nullptr));
 	}, "iconRestart");
+
+	if (Utils::FileSystem::exists("/usr/bin/odin-reboot-android"))
+	{
+		s->addEntry(_("REBOOT TO ANDROID"), false, [window] {
+			window->pushGui(new GuiMsgBox(window, _("REALLY REBOOT TO ANDROID?"),
+				_("YES"), [window] {
+					if (Utils::Platform::ProcessStartInfo("/usr/bin/odin-reboot-android --no-reboot").run() == 0)
+						Utils::Platform::quitES(Utils::Platform::QuitMode::REBOOT);
+					else
+						window->pushGui(new GuiMsgBox(window, _("COULD NOT SELECT ANDROID. REBOOT CANCELLED."), _("OK"), nullptr));
+				},
+				_("NO"), nullptr));
+		}, "iconRestart");
+	}
 
 	if (ApiSystem::getInstance()->isScriptingSupported(ApiSystem::SUSPEND))
 	{
