--- a/bootctrl_impl.c
+++ b/bootctrl_impl.c
@@ -92,32 +92,29 @@
 
 void get_kernel_cmdline_arg(const char *arg, char *buf, const char *def)
 {
-	int fd;
-	char pcmd[MAX_CMDLINE_SIZE];
-	char *val, *found, *ptr = buf;
-	fd = open("/proc/cmdline", O_RDONLY);
-	int rc = read(fd, pcmd, MAX_CMDLINE_SIZE);
-	if (rc < 0) {
-		fprintf(stderr, "Couldn't open /proc/cmdline: %d (%s)\n", rc, strerror(errno));
-		goto error;
-	}
+	char pcmd[MAX_CMDLINE_SIZE + 1] = { 0 };
+	strcpy(buf, def);
+	int fd = open("/proc/cmdline", O_RDONLY);
+	if (fd < 0)
+		return;
+	ssize_t size = read(fd, pcmd, MAX_CMDLINE_SIZE);
 	close(fd);
-	found = strstr(pcmd, arg);
-	if (!found || !(val = strstr(found, "="))) {
-		fprintf(stderr, "Couldn't find cmdline arg: '%s'\n", arg);
-		goto error;
-	}
-
-	val++;
-	// no this doesn't handle quotes lol
-	while (*val != ' ') {
-		*ptr++ = *val++;
-	}
-
-	return;
-
-error:
-	strcpy(buf, def);
+	if (size <= 0)
+		return;
+	char *save;
+	for (char *token = strtok_r(pcmd, " \n", &save); token;
+	     token = strtok_r(NULL, " \n", &save)) {
+		char *key = strchr(token, '=');
+		if (!key)
+			continue;
+		*key++ = '\0';
+		if (!strcmp(token, arg) ||
+		    (!strcmp(arg, BOOT_SLOT_PROP) && !strcmp(token, "androidboot.slot_suffix"))) {
+			strcpy(buf, key);
+			return;
+		}
+	}
+
 }
 
 // Get the value of one of the attribute fields for a partition.
@@ -342,7 +339,7 @@
 	}
 
 	for (uint32_t i = 0; i < num_slots; i++) {
-		if (get_boot_attr(&disk, i, ATTR_SLOT_ACTIVE)) {
+		if (get_boot_attr(&disk, i, ATTR_SLOT_ACTIVE) == 1) {
 			gpt_disk_free(&disk);
 			return i;
 		}
@@ -364,7 +361,7 @@
 	char bootSlotProp[MAX_CMDLINE_SIZE] = { '\0' };
 	unsigned i = 0;
 	num_slots = get_number_slots();
-	if (num_slots == 0)
+	if (num_slots != 2)
 		return -ENOENT;
 	if (num_slots == 1) {
 		// Slot 0 is the only slot around.
@@ -470,7 +467,8 @@
 
 	LOGD("Marking slot %s as active:\n", slot_suffix_arr[slot]);
 
-	for (i = 0, slotA = g_all_ptns[0]; i < ARRAY_SIZE(g_all_ptns); slotA = g_all_ptns[++i]) {
+	for (i = 0; i < ARRAY_SIZE(g_all_ptns); i++) {
+		slotA = g_all_ptns[i];
 		// Chop off the slot suffix from the partition name to
 		// make the string easier to work with.
 		LOGD("Part: %s\n", slotA);
